### PR TITLE
Extract `IGit`

### PR DESCRIPTION
**Pull request by ras0219-msft**
Picking up #529 since @vicroms does not have time to push it through.

This PR condenses all use of git to go through a narrow `IGit` interface. This hides the underlying implementation, potentially for future replacement by libgit2.

### CODE REVIEW

**Comment by BillyONeal** [comment]
>potentially for future replacement by libgit2

Considering that doesn't seem to be maintained anymore I highly doubt we would do that

**Comment by ras0219-msft** [comment]
https://github.com/libgit2/libgit2 shows the last commit at 22 days ago.

> libgit2 is used to power Git GUI clients like [GitKraken](https://gitkraken.com/) and [gmaster](https://gmaster.io/) and on Git hosting providers like [GitHub](https://github.com/), [GitLab](https://gitlab.com/) and [Azure DevOps](https://azure.com/devops). We perform the merge every time you click "merge pull request".

**Comment by BillyONeal** [comment]
It's been over 3 months, @ras0219-msft are you still working on this or should it be closed for now?

File: src/vcpkg/registries.cpp [hunk 1]
```diff
@@ -1208,6 +1308,25 @@ namespace vcpkg
             });
     }
 
+    ExpectedL<std::map<std::string, std::string>> get_builtin_port_git_objects(const VcpkgPaths& paths)
+    {
+        const auto& git = paths.get_git_impl(stdout_sink);
+        GitLsTreeOptions opts = {};
+        opts.dirs_only = true;
+        auto maybe_files = git.ls_tree(paths.git_builtin_config(), "HEAD:ports/", opts);
```
Comment (vicroms) [inline src/vcpkg/registries.cpp:1316]: End-to-end tests are failing due to this hardcoded `"HEAD:ports/"`. We should make this use the value of `--x-builtin-ports-root`.

**Comment by BillyONeal** [comment]
As this has had no work in over 6 months, I'm closing it for now. Please feel to reopen or make a new PR should you choose to pick it back up again :)

**Pull request closed without merging**